### DISCUSSION TITLE
Incorrect hover/definition for compound string key

### DISCUSSION THREAD

**Discussion by R-O-C-K-E-T** [microsoft/pylance-release#4719] [Enhancement]
## Description
If a compound string (`'prefix' 'suffix'`) is used as a key when constructing a `TypedDict` expression where the either the first string or last string section is also a valid key into the `TypedDict`, then the info/definition when hovering over those segments is incorrect.

## Expected behaviour
Either the whole compound string is highlighted and the correct type info for the whole key is shown or no type info should be shown for compound string keys.

## Steps
![Screenshot from 2023-08-06 14-46-38](https://github.com/microsoft/pyright/assets/2784602/e2d256c6-a714-4a88-8136-e80632b6d18a)

```python
class Test(TypedDict):
    prefix: str
    suffix: str
    prefixsuffix: int

def test_taker(val: Test):
    ...


test: Test = {
    'prefix': 'foo',
    'suffix': 'bar',

    'prefix' 'suffix': 30, # Hovering over 'prefix' or 'suffix' on this line produces incorrect hover text and definition location
}
test_taker({'prefix': 'foo', 'suffix': 'bar', 'prefix' 'suffix': 30})  # Again, hovering over 'prefix' or 'suffix' on this line produces incorrect hover text and definition location
```

**Comment by erictraut** [comment]
@rchiodo or @debonte, please transfer this to the pylance-relase project since it's related to a language server feature.

I don't know if this is worth addressing since it's such an edge case and the workaround is obvious — replace the compound key strings with a single string literal. I'll leave it to the pylance team to decide whether to address this.

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
